### PR TITLE
Adding templates to install cockroachdb into minishift.

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

File: cockroachdb-petset.yaml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            path: "/_admin/v1/health"
+            port: 8080
+          initialDelaySeconds: 180
+      # No pre-stop hook is required, a SIGTERM plus some time is all that's
+      # needed for graceful shutdown of a node.
+      terminationGracePeriodSeconds: 60
+      volumes:
+      - name: datadir
+        persistentVolumeClaim:
+          claimName: datadir
```
Comment (jimmidyson) [inline cockroachdb-petset.yaml]: `cockroach-datadir` perhaps?

File: cockroachdb-petset.yaml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                  },
+                  "topologyKey": "kubernetes.io/hostname"
+                }]
+              }
+            }      
+    spec:
+      containers:
+      - name: cockroachdb
+        image: redhat-ipaas/cockroachdb
+        imagePullPolicy: IfNotPresent
```
Comment (jimmidyson) [inline cockroachdb-petset.yaml]: Should leave this empty to use defaults (with preferably tagged Docker image).

File: cockroachdb-petset.yaml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                    }]
+                  },
+                  "topologyKey": "kubernetes.io/hostname"
+                }]
+              }
+            }      
+    spec:
+      containers:
+      - name: cockroachdb
+        image: redhat-ipaas/cockroachdb
```
Comment (jimmidyson) [inline cockroachdb-petset.yaml]: Can we tag this?

**Comment by jimmidyson** [comment]
Thanks for this! Have you tried it against our OSD cluster?

**Comment by chirino** [comment]
@jimmidyson no not yet.  Thought OSD was on OCP 3.3, these templates require at least OCP 3.4.

**Comment by jimmidyson** [comment]
@chirino Ah OK.  We can use PostgreSQL until we upgrade and experiment with CockroachDB a bit as we get further.

File: cockroachdb-petset.yaml (hunk 2 above)
Comment (chirino) [reply cockroachdb-petset.yaml]: will fix.  The docker image for this is at https://github.com/redhat-ipaas/cockroachdb-docker could you help me setup a CI job to get that published?

File: cockroachdb-petset.yaml (hunk 1 above)
Comment (chirino) [reply cockroachdb-petset.yaml]: The PVC's name ends up being datadir-cockroachdb-${i}. I think the PetSet handles giving it better final name.

File: cockroachdb-petset.yaml (hunk 3 above)
Comment (chirino) [reply cockroachdb-petset.yaml]: Sound good but I need help with that.

**Comment by rhuss** [comment]
@chirino  @jimmidyson  Is this PR still relevant ?

**Pull request closed without merging**